plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -192,8 +192,10 @@ considerInlineSat tm = do
               -- it would have before, but we can make it happen more often.
               -- We could potentially do this safely in non-conservative mode.
               isBindingPure <- isTermBindingPure (varStrictness varInfo) (varDef varInfo)
```

Comment:
Incidentally, this function has an extra unnecessary level of nesting. You could do:
```
let isAcceptable = ...
isBindingPure <-
if isAcceptable && isFullyApplied ... && isBindingpure
then ...
else ...
``

Reply:
Since this is monadic and thus sequential, I thought doesn't make sense to always calculate `isBindingPure`?

Reply:
I guess because this it not impure it's still gonna be lazy. I will un-nest it.